### PR TITLE
Added Hackernews Bolt

### LINKED ISSUE #23: Create benten-hacker-news bolt
Create benten-hacker-news bolt that would expose useful methods to be used in action handlers
https://github.com/HackerNews/API

### PR DESCRIPTION
**Pull request by jakeleveroni**
fixes #23 implemented the hacker news bolt with a decent amount of functionality based on its very limited API.

### CODE REVIEW

**Comment by jakeleveroni** [comment]
The changes to the following files were unintentional but im not sure how or why they were changed in the first place:

BenetenStarterApplication.java
BenetenApplication.java
HtmlToImageConverter.java
AiException.java
ImageGenerationExcaption.java
AiProperties.java
BentenProxyConfig.java
SlackProperties.java
NlpClient.java
DialogFlowClient.java
DialogFlowClientTest.java

Im thinking maybe maven commands updated the files when i didnt want them to. @DivakarUngatla  do you have any idea on how i can revert these other than manually?

**Comment by DivakarUngatla** [comment]
Hi @jakeleveroni Jacob,

Looks like you have duplicated generic property files like AiProperties , BentenProxyConfig.java etc.. in the hacker rank bolt that you created (this is not necessary, you should only have HackerNewsProperties. java in you hacker rank bolt). And all references to these property files have changed to the property files in hackerbolt (this could have happened because of an IDE auto import).

I would recommend create a new branch from master in your for and then use [git cherrypick](https://www.atlassian.com/git/tutorials/cherry-pick) to transfer only necessary commits from the current branch to new branch and raise a PR from the new branch.

File: benten-hackernews-bolt/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/target/
```
Comment (jakeleveroni) [inline benten-hackernews-bolt/.gitignore]: Other bolts included their target dir, do you want mew to include the built bolt as well?

Comment (DivakarUngatla) [reply benten-hackernews-bolt/.gitignore]: @jakeleveroni this is fine, the right thing to do. I will exclude this for other bolts as well.

File: benten-core/src/main/java/com/intuit/benten/properties/BentenProxyConfig.java [hunk 2]
```diff
@@ -56,6 +56,6 @@ public String getProtocol() {
     public void setProtocol(String protocol) {
         this.protocol = protocol;
     }
-
```
Comment (DivakarUngatla) [inline benten-core/src/main/java/com/intuit/benten/properties/BentenProxyConfig.java]: sorry for being picky but can you revert this change as there is no change except an extra space ?

File: benten-hackernews-bolt/README.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+ 
+1. `cd ~/dev/benten/hackernews-bolt;`
+2. `mvn clean install`
+3. `cd ~/dev/benten/benten-starter;`
+5. `mvn clean install -Dmaven.test.skip=true`
+6. `mvn spring-boot:run;`
+
+If you get an error about the `benten-hackernews-bolt` not being found, try running this:
+
+`cd ~/dev/benten/benten-hackernews-bolt/; mvn install:install-file -Dfile=/Users/jleveroni/dev/benten/benten-hackernews-bolt/target/benten-hackernews-bolt-0.1.5.jar -DgroupId=benten-hackernews-bolt -DartifactId=benten-hackernews-bolt -Dversion=0.1.5 -Dpackaging=jar -DgeneratePom=true;`
```
Comment (DivakarUngatla) [inline benten-hackernews-bolt/README.md]: You can remove this section. Since you added packaging as jar a mvn clean install in the parent benten directory should generate the artifact and install it in the m2 repository. Else you can replace specific path variables with place holders.

File: benten-hackernews-bolt/README.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
+# Hackernews Bolt
```
Comment (DivakarUngatla) [inline benten-hackernews-bolt/README.md]: If possible can you include a zip version of the dialog flow with the actions you have implemented ? You can download it from dialog flow as a zip file and upload it here https://github.com/DivakarUngatla/divakarungatla.github.io/raw/master/benten/dialogflow-agent and provide this link in the Readme.

Comment (jakeleveroni) [reply benten-hackernews-bolt/README.md]: i used the same project that you had listed in the setup. Just added some new intents do you want me to commit a second file or overwrite the one you have in that repo? also you may need to open that repo up i was unable to commit to it.

Comment (DivakarUngatla) [reply benten-hackernews-bolt/README.md]: Can you commit it as a new file and name it as benten-open-source-hackernews-bolt ? You can raise a PR against that repo similar to how you did it here. I will merge it.
